tests/integration/awslambda/test_lambda_integration.py

```diff
@@ -422,6 +426,174 @@ def verify_failure_received():
+                {"id": {"S": "test1234"}, "numericFilter": {"N": "12"}},
+                {"dynamodb": {"NewImage": {"numericFilter": {"N": [{"numeric": [">", 100]}]}}}},
+                0,
+            ),
+            (
+                {"id": {"S": "test123"}, "numericFilter": {"N": "100"}},
+                {"id": {"S": "test1234"}, "numericFilter": {"N": "12"}},
+                {
+                    "dynamodb": {
+                        "NewImage": {"numericFilter": {"N": [{"numeric": [">=", 100, "<", 200]}]}}
+                    }
+                },
+                0,
+            ),
+            # Prefix
+            (
+                {"id": {"S": "test123"}, "prefix": {"S": "us-1-testtest"}},
+                {"id": {"S": "test1234"}, "prefix": {"S": "testtest"}},
+                {"dynamodb": {"NewImage": {"prefix": {"S": [{"prefix": "us-1"}]}}}},
+                1,
+            ),
+        ],
+    )
+    def test_dynamo_db_insert_event(
+        self,
+        create_lambda_function,
+        lambda_client,
+        dynamodb_client,
+        dynamodb_create_table,
+        lambda_su_role,
+        logs_client,
+        wait_for_dynamodb_stream_ready,
+        filter,
+        calls,
+        item_to_put1,
+        item_to_put2,
+    ):
+
+        function_name = f"lambda_func-{short_uid()}"
+        table_name = f"test-table-{short_uid()}"
+        event_source_uuid = None
+        max_retries = 50
+
+        try:
+
+            create_lambda_function(
+                handler_file=TEST_LAMBDA_PYTHON_ECHO,
+                func_name=function_name,
+                runtime=LAMBDA_RUNTIME_PYTHON37,
+                role=lambda_su_role,
+            )
+            dynamodb_create_table(table_name=table_name, partition_key="id")
+            _await_dynamodb_table_active(dynamodb_client, table_name)
+            stream_arn = dynamodb_client.update_table(
+                TableName=table_name,
+                StreamSpecification={"StreamEnabled": True, "StreamViewType": "NEW_AND_OLD_IMAGES"},
+            )["TableDescription"]["LatestStreamArn"]
+            wait_for_dynamodb_stream_ready(stream_arn)
+            event_source_mapping_kwargs = {
+                "FunctionName": function_name,
+                "BatchSize": 1,
+                "StartingPosition": "TRIM_HORIZON",
+                "EventSourceArn": stream_arn,
+                "MaximumBatchingWindowInSeconds": 1,
+                "MaximumRetryAttempts": 1,
+            }
+            event_source_mapping_kwargs.update(
+                FilterCriteria={
+                    "Filters": [
+                        {"Pattern": json.dumps(filter)},
+                    ]
+                }
+            )
+            try:
+                event_source_uuid = lambda_client.create_event_source_mapping(
+                    **event_source_mapping_kwargs
+                )["UUID"]
+            except Exception as ex:
```

Comment:
I would suggest the following:
- remove this section with try-except
- remove the first parameterized test-values with the `None` filter
- add a new test case that tests invalid filter, eg. `test_dynamodb_invalid_event_filter`

In this new test, you can some invalid filter-pattern, and then use the `pytest.raises` pattern. I already tested the following:

```
@pytest.mark.aws_validated
@pytest.mark.parametrize(
        "filter",
        [
            '[{"eventName": ["INSERT"=123}]',
          'null'
        ]
    )
   def test_dynamodb_invalid_event_filter(self, create_lambda_function,
        lambda_client,
        dynamodb_client,
        dynamodb_create_table,
        lambda_su_role,
        wait_for_dynamodb_stream_ready,
                                           filter):
        # do the setup like in the other test
        event_source_mapping_kwargs.update(
            FilterCriteria={
                "Filters": [
                    {"Pattern": filter},
                ]
            }
        )
        with pytest.raises(Exception) as expected:
            lambda_client.create_event_source_mapping(
                **event_source_mapping_kwargs
            )
        assert expected.typename == "InvalidParameterValueException"
```